graphics.tex

```diff
@@ -132,17 +129,188 @@ \subsection*{Hires}
+		Result & 11100000 & 224\\
+		\hline
+	\end{tabular} \\
+\end{tabular}
+
+You can test this, by the way, with some simple PRINT commands:
+
+\begin{screenoutput}
+PRINT 224 AND 127
+  96
+
+PRINT 96 OR 128
+  224
+\end{screenoutput}
+
+Armed with this knowledge, we can now start configuring the VIC to show bitmap graphics. For this we have to fine-tune a number of registers but before we can start with that, we need to understand a bit more about the graphics chip, how it is organized and most importantly how it accesses the memory.
+
+\subsection*{VIC IV organization and memory access basics}
+
+In contrast to the cpu of your MEGA65, which can address 64KB at once, the VIC unfortunately is not able to this. It can only see 16KB, so you have to define exactly which 16KB it is allowed to see. To achieve this, the 64KB of Bank 0 (this is the memory bank where VIC is living and getting ins data) is divided into blocks of 16KB each. These 16KB blocks are called "video banks" or "VIC banks".
+
+\includegraphics[width=\linewidth]{images/graphics/vic-banks.png}
+
+Independent of the VIC, the memory management of the Commodore 65 or MEGA65 is somewhat more complex than that of the previous model. This is because there is more memory available in the machine: For the Commodore 65 128KB where planned and the MEGA65 comes with 384KB which can be scaled up to max 1 MB. As already mentioned, the processor is not able to address the whole range of memory. Only 64KB can be seen. For this reason the memory is divided into blocks, 64KB each, and these blocks are called "banks". The system can have up to 16 banks and certain banks also have certain tasks. Bank 0 for example is called the system bank, coming from BASIC65, this is the bank where VIC lives. The same applies to the screen RAM, which occupies the memory range \$0800-\$2000 after switching the computer on. This is also in bank 0. Banks 0 and 1 are RAM, which means you can read and write to them, while banks 2 and 3 are ROM and not so easy to override. By the way, the BASIC65 implementation itself is stored in the banks listed last. If you want to know more about memory management in general please have a look at Appendix F (System Memory Map). In BASIC - sorry to say - the situation is getting even more complex. Right after turning on the computer you are in Bank 128 which is a special bank with a special memory configuration. You can use the command BANK to switch between banks. A problem here is, that you can easily lose track of what is in which bank or in which bank you are currently working. It's actually a bit easier when working in machine language. For more information about the BANK-command and banks in special, please look into Appendix B (BASIC65 Command Reference). Please keep in mind that you need to know about banks, before you can safely use PEEK, POKE or SYS. Otherwise you may wonder why your program does not behave as expected. We will come back to this later. Back to graphics. 
```

Comment:
General thought: This is quite a lengthy paragraph, wondering if it's worth breaking it up into a few paragraphs.

Think it should read:
`For the Commodore 65, 128KB was planned (potentially expandable to 1MB), while the MEGA65 comes with 384KB of chip RAM + 8MB of attic RAM (potentially expandable to 256MB)`

Maybe I'd change the ordering of your bank descriptions to emphasise first-and-foremost that bank 0 and 1 are RAM (before getting into the gory details of what is within each bank). E.g.:

> The system can have up to 16 banks and certain banks also have certain tasks.
> Banks 0 and 1 are RAM, which means you can read and write to them, while banks 2 and 3 are ROM and not so easy to override.
> Bank 0 (from the perspective of the BASIC 65 operating system that runs when you power the system on) is call the system bank. It contains regions such as:
> - the screen RAM, which occupies the memory range \$0800-\$2000
> - memory-mapped VIC registers
> By the way, the BASIC 65 implementation itself is stored in ROM (bank 3), while the C64's BASIC is also stored in ROM (bank 2).

Change to `please have a look at Appendix \bookvref{cha:memory-map}.`

Change to `the situation gets even more complex.`

Hmm, Bank 128 is a concept that is pretty unique to BASIC 65 itself and doesn't really relate to the 128th bank in the addressable memory space. I wonder if it's worth highlighting/emphasising that (though I guess maybe they can garner that info from your suggestion to look at Appendix B - BANK).

Change Appendix B to `\bookvref{BASIC 65 Commands!BANK}`.